dbt/tests/generic/test_res_class_matches_pardat.sql

```diff
@@ -3,17 +3,31 @@
 -- filters for residential parcels by anti-joining the model against `comdat`
 -- using parid and taxyr; as a result, it filters out mixed-use parcels as well.
 --
--- By default, the test will compare the first 3 digits of each set of classes;
--- if `major_class_only=true`, however, the test will compare the first digit
--- only.
+-- Optional parameters:
+--
+--   * major_class_only (bool): Compare only the first digit of classes. When
+--         set to False, compare the first three digits instead.
+--   * parid_column_name (str): The name of the column on the base model that
+--         corresponds to `parid`, in case the model uses a different name scheme.
+--   * taxyr_column_name (str): The name of the column on the base model that
+--         corresponds to `taxyr`, in case the model uses a different name scheme.
+--   * additional_select_columns (dict): Standard parameter for selecting
+--         additional columns from the base model for use in reporting failures.
+--         `model.{column_name}` and `pardat.class` are already selected by
+--         default and do not need to be included using this parameter.
+--         See the `format_additional_select_columns` macro for more information.
+--         Note that while this parameter can often be either a string or a dict,
+--         a dict is required for this test and it must contain the `alias`
+--         and `agg_func` parameters given that results are grouped by `parid`
+--         and `taxyr`.
 --
--- Since the output is grouped by (parid, taxyr), additional columns that would
--- normally be selected via `additional_select_columns` need to be selected
--- with one of two aggregation methods: `select_columns_aggregated_with_array`
--- (which uses the `array_agg()` function to select the columns) or
--- `select_columns_aggregated_with_max` (which uses the `max()` function).
 {% test res_class_matches_pardat(
-    model, column_name, major_class_only=false, additional_select_columns=[]
+    model,
+    column_name,
+    major_class_only=false,
+    parid_column_name="parid",
+    taxyr_column_name="taxyr",
```

Comment:
Note that this join interface is less flexible than the one implemented on `row_values_match_after_join` (where the join is specified via a raw SQL string passed to a `join_condition` parameter), but I think that's acceptable for now since this test is more specific than `row_values_match_after_join` and hence can afford to be less flexible for the sake of a simpler interface.